Digipost/Logger.swift

```diff
@@ -25,39 +25,50 @@ class Logger {
+    Example on format with a single String:
+    Logger dpostLogError:@"Description: State-parameter returned from server differed from what client sent | Location: Showing OAuth Login Screen | UI: User will get an error and be asked to try logging in again | Cause: Server is broken or man in the middle"];
+
+    :param: description    Explanation on what caused the error
+    :param: location       Where in the app the error happened
+    :param: UI             How will it look to the user
+    :param: cause          What most likely caused the error
+    */
+    class func dpostLogError(description: String, location: String, UI: String, cause: String ) {
+        let message = standardizedMessage(description, location: location, UI: UI, cause: cause)
+        dpostLog(.Error, message: message)
     }
 
-    class func dpostLogInfo(message: String) {
-        dpostLog(DpostLogSeverity.Info, message: message)
+    /**
+    Log a warning to digipost log service on a specific format.
+    Example on format with a single String:
+    Logger dpostLogWarning:@"Description: State-parameter returned from server differed from what client sent | Location: Showing OAuth Login Screen | UI: User will get an error and be asked to try logging in again | Cause: Server is broken or man in the middle"];
+
+    :param: description    Explanation on what caused the warning
+    :param: location       Where in the app the error happened
+    :param: UI             How will it look to the user
+    :param: cause          What most likely caused the warning
+    */
+    class func dpostLogWarning(description: String, location: String, UI: String, cause: String ) {
+        let message = standardizedMessage(description, location: location, UI: UI, cause: cause)
+        dpostLog(.Warn, message: message)
     }
 
-    class func dpostLogWarn(message: String) {
-        dpostLog(DpostLogSeverity.Warn, message: message)
-    }
-
-    class func dpostLogError(message: String) {
-        dpostLog(DpostLogSeverity.Error, message: message)
+    private class func standardizedMessage(description: String , location: String, UI: String, cause: String) -> String {
+        return "Description: \(description) | Location: \(location) | UI: \(UI) | Cause: \(cause)"
     }
 
     private class func dpostLog(severity: DpostLogSeverity, message: String) {
-        if Logger.Constants.currentlyEnabled {
-            let parameters = [ Logger.Constants.severity : severity.rawValue, Logger.Constants.message : message ]
-            APIClient.sharedClient.postLog(uri: Logger.Constants.uri, parameters: parameters)
-        }
-        DLog(message)
+        let parameters = [ Logger.Constants.severity : severity.rawValue, Logger.Constants.message : message ]
+        APIClient.sharedClient.postLog(uri: Logger.Constants.uri, parameters: parameters)
     }
 
-
-
 }
 
- func DLog(message: String, function: String = __FUNCTION__) {
+func DLog(message: String, function: String = __FUNCTION__) {
```

Comment:
Maybe we should rename from DLog to `debugLog` or something like that? Maybe `localDebugLog`…